src/components/network-chart/index.js

```diff
@@ -0,0 +1,235 @@
+
+    if (shortUriIndex + 1 < uri.length) {
+      const shortUri = uri.slice(shortUriIndex, uri.length);
+      return shortUri;
+    }
+    return uri;
+  }
+
+  render() {
+    const { marker, markerStyle } = this.props;
+    const itemClassName = ('item ' + marker.data.status).toLowerCase();
+
+    return (
+      <section className={itemClassName}>
+        <div className="status" />
+        <div className="itemLabel" title={marker.data.URI}>
+          {this._shortenURI(marker.data.URI)}
+        </div>
+        <div
+          className="itemBar"
+          style={markerStyle}
+          onMouseEnter={this._hoverIn}
+          onMouseLeave={this._hoverOut}
+        >
+          &nbsp;
+        </div>
+        {this.state.hovered ? (
+          <Tooltip mouseX={this.state.pageX} mouseY={this.state.pageY}>
+            <MarkerTooltipContents
+              marker={marker}
+              threadIndex={this.props.threadIndex}
+            />
+          </Tooltip>
+        ) : null}
+      </section>
+    );
+  }
+}
+
+class NetworkChart extends React.PureComponent<Props> {
+  constructor(props: any) {
+    super(props);
+    (this: any)._renderRow = this._renderRow.bind(this);
+    (this: any)._getMarkerLength = this._getMarkerLength.bind(this);
+    (this: any)._getMarkerPosition = this._getMarkerPosition.bind(this);
+    (this: any)._getMarkerStyling = this._getMarkerStyling.bind(this);
+  }
+
+  /**
+   * Determine the maximum zoom of the viewport.
+   */
+  getMaximumZoom(): UnitIntervalOfProfileRange {
+    const { timeRange: { start, end }, interval } = this.props;
+    return interval / (end - start);
+  }
+
+  _getMarkerPosition(markerStart) {
+    const timeRange = this.props.timeRange;
+    const timeRangeTotal = timeRange.end - timeRange.start;
+
+    return (markerStart - timeRange.start) * 75 / timeRangeTotal;
+  }
+
+  _getMarkerLength(markerDuration) {
+    const timeRange = this.props.timeRange;
+    const timeRangeTotal = timeRange.end - timeRange.start;
+
+    let markerLength = markerDuration / timeRangeTotal * 75;
+
+    if (markerLength < 0.15) {
+      markerLength = 0.15;
+    }
+
+    return markerLength;
+  }
+
+  _getMarkerStyling(marker) {
+    const markerLength = this._getMarkerLength(marker.dur);
+    const markerPosition = this._getMarkerPosition(marker.data.startTime);
+
+    const markerStyling = {
+      width: markerLength + '%',
+      left: markerPosition + '%',
+    };
+    return markerStyling;
+  }
+
+  // Create row with correct details
+  _renderRow(nodeId: any, index: number) {
+    const markersPerRange = this._filterMarkersPerRange(
+      this.props.timeRange,
+      this.props.markers
+    );
+
+    return (
+      <NetworkChartRow
+        marker={markersPerRange[index]}
+        index={index}
+        markerStyle={this._getMarkerStyling(markersPerRange[index])}
+        threadIndex={this.props.threadIndex}
+      />
+    );
+  }
+
+  _filterMarkersPerRange(timeRange: Object, markers: []) {
```

Comment:
the typing here doesn't look right ?

But anyway you'll likely need to remove it and use a selector instead, see the more complete comment below.

Reply:
You stated that you are still working on types, but I'll add a few comments here anyway.

`Object` is a small step above an `any` type, as it doesn't add much type refinement. Also, `[]` indicates an empty array, as opposed to some kind of generic list of things. The existential type `*` can help indicate to have flow figure out what is going on.